numpy/lib/index_tricks.py

```diff
@@ -769,9 +769,9 @@ def fill_diagonal(a, val, wrap=False):
     a : array, at least 2-D.
       Array whose diagonal is to be filled, it gets modified in-place.
 
-    val : scalar
-      Value to be written on the diagonal, its type must be compatible with
-      that of the array a.
+    val : scalar or array_like
+      Value or values to write on the diagonal; type must be compatible with that of array a. If an array of values, the length of values must be equal to the 
+      main diagonal to fill.
```

Comment:
```suggestion
    val : scalar or array_like
      Value(s) to write on the diagonal. If `val` is scalar, the value is
      written along the diagonal. If array-like, the flattened `val` is
      written along the diagonal, repeating if necessary to fill all
      diagonal entries.
```
Here's a quick wording suggestion - feel free to ignore/modify as you see fit.

Reply:
I'll take the experts explanation over my own. If everyone agrees on it then I have no comments!